.github/workflows/ci.yml

```diff
@@ -51,3 +51,30 @@ jobs:
       - name: Generate FFI bindings
         run: just gen
       - run: cargo clippy --all-targets -- -D warnings
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: extractions/setup-just@v1
+      - name: Setup rust toolchain
+        run: rustup show
+      - uses: Swatinem/rust-cache@v2.2.0
+      - uses: subosito/flutter-action@v2
+        with:
+          channel: "stable"
+      - name: Install FFI bindings
+        run: cargo install flutter_rust_bridge_codegen
+      - name: Start containers
+        run: |
+          docker-compose up -d
+          sleep 10 # We need to give docker a bit of time to startup
+
+      - name: Test containers are up
+        run: |
+          curl -d '{"address":"bcrt1qylgu6ffkp3p0m8tw8kp4tt2dmdh755f4r5dq7s", "amount":"0.1"}' -H "Content-Type: application/json" -X POST http://localhost:3000/faucet
```

Comment:
🤔 How about adding a retry logic here? by checking if the returned `http_code` equals `200`

```
$(curl -LI http://google1.com -o /dev/null -w '%{http_code}\n' -s)
```

This might also allow us to remove the `sleep 10` from the step before.